mkosi/__init__.py

```diff
@@ -910,7 +918,7 @@ def save_cache(state: MkosiState) -> None:
             shutil.move(state.build_overlay, build)
 
 
-def dir_size(path: PathString) -> int:
+def dir_size(path: Path) -> int:
```

Comment:
Let's accept `os.DirEntry` here as well, saves the unnecessary cast further down and since this is a recursive function, the call is paid multiple times.